### PR TITLE
hardening/54_add_app_utils_and_mysql_driver_tests

### PR DESCRIPTION
**Pull request by frbattid**
* Partially implements the issue #54 
* 100% unit tests passed:
```
***** STARTING TESTS *****

  ․ [appUtils.buildUsername] build a username from a valid base string should return frb1 when frb is used as base string: 0ms
    [appUtils.buildUsername] build an invalid username from an invalid base string should return null when fiware is used as base string: error: The base username "fiware" is not allowed
  ․ [appUtils.buildUsername] build an invalid username from an invalid base string should return null when fiware is used as base string: 2ms
    [appUtils.provisionCluster] provision a cluster should redirect to /after when being called from /before: info: Successful command executed: 'bash -c echo "sudo useradd frb" | ssh -i ./priv_key admin@fake.cosmos.lab.fiware.org'
info: Successful command executed: 'bash -c echo "echo 12345 | sudo passwd frb --stdin" | ssh -i ./priv_key admin@fake.cosmos.lab.fiware.org'
info: Successful command executed: 'bash -c echo "sudo -u hdfs hadoop fs -mkdir /user/frb" | ssh -i ./priv_key admin@fake.cosmos.lab.fiware.org'
info: Successful command executed: 'bash -c echo "sudo -u hdfs hadoop fs -chown -R frb:frb /user/frb" | ssh -i ./priv_key admin@fake.cosmos.lab.fiware.org'
info: Successful command executed: 'bash -c echo "sudo -u hdfs hadoop fs -chmod -R 740 /user/frb" | ssh -i ./priv_key admin@fake.cosmos.lab.fiware.org'
info: Successful command executed: 'bash -c echo "sudo -u hdfs hadoop dfsadmin -setSpaceQuota 5g /user/frb" | ssh -i ./priv_key admin@fake.cosmos.lab.fiware.org'
  ․ [appUtils.provisionCluster] provision a cluster should redirect to /after when being called from /before: 2ms

  3 passing (11ms)


    [mysqlDriver.connect] create a MySQL connection should return null error: info: Connected to http://:3306/cosmos
  ․ [mysqlDriver.connect] create a MySQL connection should return null error: 3ms
    [mysqlDriver.addUser] add a new user should return null error and an empty result set: info: Successful insert: 'INSERT INTO cosmos_user (idm_username, username, password, hdfs_quota) VALUES (frb@tid.es, frb1, 12345, 5)'
  ․ [mysqlDriver.addUser] add a new user should return null error and an empty result set: 0ms
  ․ [mysqlDriver.addPassword] add a new password should return null error and an empty result set: 0ms
  ․ [mysqlDriver.getUser] get a user by the idm user should return null error and a result set containing frb1: 0ms
  ․ [mysqlDriver.getUserByCosmosUser] get a user by the cosmos user should return null error and an result set containing frb1: 0ms
  ․ [mysqlDriver.close] close a connection should finish: 0ms

  6 passing (10ms)

****** TESTS ENDED *******
```
* No e2e tests are required:
* Assignee @gtorodelvalle

### CODE REVIEW

File: cosmos-gui/test/node/app_utils_test.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        });
+    });
+});
+
+describe('[appUtils.provisionCluster] provision a cluster', function() {
+    it('should redirect to ' + pathAfter + ' when being called from ' + pathBefore, function () {
+        appUtils.provisionCluster(res, clusterPrivKey, clusterUser, clusterEndpoint, hdfsSuperuser, hdfsQuota, username, password);
+        assert.equal(pathAfter, res.url);
+    });
+});
```
Comment (gtorodelvalle) [inline cosmos-gui/test/node/app_utils_test.js]: Final blank line appreciated :)

File: cosmos-gui/test/node/mysql_driver_test.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+// Mock the mysql connection
+var connectionMock = {
+    connect: function(callback) {
+        return callback(null);
+    },
+    query: function(query, values, callback) {
+        query = query || '';
+        values = values || [];
+        callback = callback || function(error, result) {};
```
Comment (gtorodelvalle) [inline cosmos-gui/test/node/mysql_driver_test.js]: He, he :) The other option is to check `if (callback)` and only calling it if it exists :) Not to create an empty function :) What do you think?

File: cosmos-gui/test/node/mysql_driver_test.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            assert.equal(username, result[0]);
+        });
+    });
+});
+
+describe('[mysqlDriver.close] close a connection', function() {
+    it('should finish', function() {
+        mysqlDriver.close(connectionMock);
+    });
+});
```
Comment (gtorodelvalle) [inline cosmos-gui/test/node/mysql_driver_test.js]: Final blank line appreciated :)

File: cosmos-gui/test/node/mysql_driver_test.js (hunk 2 above)
Comment (frbattid) [inline cosmos-gui/test/node/mysql_driver_test.js]: The problem was not with the invocation of the callback function, but with the number of arguments of the query redefinition. When calling this mock query function with less than 3 arguments then I got an "undefined value" error.

File: cosmos-gui/test/node/mysql_driver_test.js (hunk 3 above)
Comment (frbattid) [inline cosmos-gui/test/node/mysql_driver_test.js]: Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/76f9443cc2c3e45f5d3740772e50ba7266ef6fbb

File: cosmos-gui/test/node/app_utils_test.js (hunk 1 above)
Comment (frbattid) [inline cosmos-gui/test/node/app_utils_test.js]: Fixed in https://github.com/telefonicaid/fiware-cosmos/commit/76f9443cc2c3e45f5d3740772e50ba7266ef6fbb

File: cosmos-gui/test/node/mysql_driver_test.js (hunk 2 above)
Comment (gtorodelvalle) [inline cosmos-gui/test/node/mysql_driver_test.js]: NTC :)

**Comment by gtorodelvalle** [comment]
LGTM and merging :)

**Pull request merged**